src/main/java/com/ibm/crypto/plus/provider/XDHPrivateKeyImpl.java

```diff
@@ -149,6 +155,11 @@ public XDHPrivateKeyImpl(OpenJCEPlusProvider provider, AlgorithmParameterSpec pa
                 int encodingSize = CurveUtil.getDEREncodingSize(curve);
                 this.xecKey = XECKey.createPrivateKey(provider.getOCKContext(), der, encodingSize);
             }
+            if (Integer.parseInt(provider.getVersionStr()) <= 11) {
+                super.key = this.key;
+            } else {
+                super.key = new DerValue(DerValue.tag_OctetString, this.key).toByteArray();
```

Comment:
Similar to above.

Reply:
I did not change according to above, because we still need to assign `this.key = k;` in JDK11 and before version.